hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -575,6 +576,8 @@ private BitSet setMatchedPartitions(String partitionIds, Expression filter,
    */
   public BlockMappingVO getBlockRowCount(Job job, CarbonTable table,
       List<PartitionSpec> partitions) throws IOException {
+    // no useful information for count star query without filter, so disable explain collector
+    ExplainCollector.remove();
```

Comment:
I think this modification just try to avoid the problem but don't actually solve the problem.
Can you explain what is the root cause of that problem?

Reply:
You are right. 
Normal query flow goes to `CarbonInputFormat#getPrunedBlocklets` and initialize the pruning info for table we queried.  But count star query without filter use a different query plan, it does not go into that method, so no pruning info does not init. When it calls default data map to prune(with a null filter), exception will occur during settingg pruning info.

One solution is to init the pruning info for this type of query in mrthod `getBlockRowCount`. But considering
no useful information about block/blocklet pruning for such query(actually no pruning), I choose to disable the expalin collector instead.

Reply:
That's the weird part -- We are trying to remove the pruning collector even the pruning info is not initialized.
I think you can add a flag for the collector to identify whether it is initialized. And this flag is used where carbon what to record the info. If you are planing to work like this, please add a comment for the scenario of this variable.

Reply:
No remove. Its implementation is disable.

Reply:
oh... I understand.
The current implementation of pruning collector may has bugs. Based on the current implementation, your modification is OK...

Reply:
please add comments for your modification in the code for better understanding

Reply:
OK